src/simulation/elements/PSNS.cpp

```diff
@@ -70,6 +72,43 @@ int Element_PSNS::update(UPDATE_FUNC_ARGS)
 					}
 				}
 	}
+	else if (parts[i].tmp == 1) {
+		parts[i].life = 0;
+		for (rx = -2; rx < 3; rx++)
```

Comment:
This entire for loop is completely unnecessary. You are looping through particles just to set photonWl to the current pressure.